### PR TITLE
fix(logging): log less

### PR DESCRIPTION
**Pull request by aborsu**
When overriding a key in the default config,
only log in debug if the default value is not None.
Also if the type of the values is different, warn the user.

### CODE REVIEW

File: src/__init__.py [hunk 1]
```diff
@@ -16,10 +16,15 @@ def merge_configs(a, b):
         if isinstance(a[key], dict) and isinstance(b[key], dict):
             new_config[key] = merge_configs(a[key], b[key])
         else:
-            log.warn(
-                'Overriding keys with different types %s and %s',
-                type(a[key]), type(b[key]))
             new_config[key] = b[key]
+            if a[key] is None:
+                pass
```
Comment (SagaStalin) [inline src/__init__.py]: MAJOR Either remove or fill this block of code. [rule](https://sonar.sagacify.com/coding_rules#rule_key=python%3AS108)

**Comment by SagaStalin** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by aborsu** [comment]
[xs:(config-py) log less](https://trello.com/c/G3tFNcwD/429-xs-config-py-log-less)

**Review by kfrancoi — APPROVED** [review]

**Review by satblip — APPROVED** [review]
LGTM

**Pull request merged**